pandas-stubs/core/frame.pyi

```diff
@@ -799,23 +799,53 @@ class DataFrame(NDFrame, OpsMixin, _GetItemHack):
     @overload
     def replace(
         self,
-        to_replace=...,
-        value: Scalar | NAType | Sequence | Mapping | Pattern | None = ...,
+        to_replace: (
+            Scalar
+            | NAType
+            | Sequence[Scalar | Pattern]
+            | Mapping[Hashable, Scalar | Pattern]
+            | Pattern
+            | None
+        ) = ...,
+        value: (
+            Scalar | NAType | Sequence[Scalar] | Mapping[Hashable, Scalar] | None
+        ) = ...,
```

Comment:
`value` was originally annotated to include `Pattern`, but is that right?

It can accept regex strings, sure:
```python
df = pd.DataFrame({'a': ['a', 'ab', 'ba']})
print(df.replace(f'^a(.*)', r'\1', regex=True))
```

But trying to use `df.replace(f'^a(.*)', re.compile(r'\1'), regex=True)` raises, and it doesn't seem to useful to use `re.compile` here if we can't include captured groups in `value`

There's no tests which pass a compiled regex to `value`. I think it's OK to remove the `Pattern` annotation from `value`?

Reply:
Yes.  This was something that was in the Microsoft stubs that we copied over.